docs/guides/gitlab.md

```diff
@@ -0,0 +1,280 @@
+!!!include(install-mkcert.md)!!!
+
+1.  Create a certificate for the domain that will be used for the GitLab route. For example:
+
+    ```bash
+    mkcert "gitlab.pomerium.localhost.io"
+    ```
+
+1.  Note the location of the mkcert root certificate files with `mkcert -CAROOT`. You will need to provide this path to your Pomerium configuration to validate the certificate provided by GitLab.
+::::
+
+If you have an internal certificate solution, generate a certificate for `gitlab.pomerium.localhost.io` and note the path to the certificate authority (**CA**) root before proceeding. 
+
+1.  Create the directory `/srv/gitlab/config/ssl/` (adjusted for your local Docker volume path), and move the certificate and key files there:
+
+    ```bash
+    mkdir /srv/gitlab/config/ssl
+    mv gitlab.pomerium.localhost.io.pem gitlab.pomerium.localhost.io-key.pem /srv/gitlab/config/ssl
+    ```
+
+    **Note:** You will need elevated permissions (sudo or root access) regardless of the directory location, as GitLab restricts permissions from within the Docker container on the volume mount.
+
+1.  Adjust the `docker-compose.yml` entry for Pomerium to mount the internal certificate authority, and the entry for GitLab to specify the certificate and key files:
+
+    ```yaml
+    services:
+
+    ...
+
+      pomerium:
+        image: pomerium/pomerium:latest
+        container_name: pomerium
+        volumes:
+          - ./srv/pomerium/config.yaml:/pomerium/config.yaml:ro
+          - ~/.local/share/mkcert:/pomerium/ssl:ro # Adjust to the location of your internal certificate authority.
+        ports:
+          - 443:443
+          - 80:80
+    ...
+
+    gitlab:
+      container_name: gitlab
+      image: gitlab/gitlab-ee:latest
+      environment:
+        GITLAB_OMNIBUS_CONFIG: |
+        external_url 'https://gitlab.pomerium.localhost.io'
+        registry_external_url 'https://gitlab.pomerium.localhost.io'
+        letsencrypt['enable'] = false
+        nginx['listen_port'] = 443
+        nginx['listen_https'] = true
+        nginx['ssl_certificate'] = "/etc/gitlab/ssl/gitlab.localhost.pomerium.io.pem"
+        nginx['ssl_certificate_key'] = "/etc/gitlab/ssl/gitlab.localhost.pomerium.io-key.pem"
+        registry_nginx['listen_port'] = 443
+        registry_nginx['listen_https'] = true
+        registry_nginx['ssl_certificate'] = "/etc/gitlab/ssl/gitlab.localhost.pomerium.io.pem"
+        registry_nginx['ssl_certificate_key'] = "/etc/gitlab/ssl/gitlab.localhost.pomerium.io-key.pem"
+        pages_nginx['listen_port'] = 443
+        pages_nginx['listen_https'] = true
+        pages_nginx['ssl_certificate'] = "/etc/gitlab/ssl/gitlab.localhost.pomerium.io.pem"
+        pages_nginx['ssl_certificate_key'] = "/etc/gitlab/ssl/gitlab.localhost.pomerium.io-key.pem"
+        mattermost_nginx['listen_port'] = 443
+        mattermost_nginx['listen_https'] = true
+        mattermost_nginx['ssl_certificate'] = "/etc/gitlab/ssl/gitlab.localhost.pomerium.io.pem"
```

Comment:
```suggestion
        mattermost_nginx['ssl_certificate'] = "/etc/gitlab/ssl/ mattermost.localhost.pomerium.io.pem"
```

I'm not sure how things like pages (i assume like github pages) and mattermost are expected to work or if they should be on different domains (and need separate pomerium routes).

Reply:
I added this because in your last review you modified pages and mattermost settings, which I had not touched. What if we removed them from this section and [above](https://github.com/pomerium/pomerium/pull/2800/files#diff-4c55b536664cdc4b47e8d0b3bc1406046044f611716bb7cc48662aa02e70b92aR70-R75), and add a note that additional integrations may  require additional custom settings?

Reply:
Staged in 9a630b9dc4bb38c39fa477d522321575ca33e1df